pkg/api/api.go

```diff
@@ -669,6 +682,28 @@ func (a *StreamplaceAPI) HandleLivestream(ctx context.Context) httprouter.Handle
 	}
 }
 
+func (a *StreamplaceAPI) RateLimitMiddleware(ctx context.Context) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			// XXX: check if x-forwarded-for
```

Comment:
not a concern for streamplace in production, we host our own HTTPS, but not impossible someone else could configure a proxy for sure